### PR TITLE
prov/shm: Add support for write() method to submit DSA work and changes to page fault handling

### PR DESCRIPTION
**Pull request by jcabr020**
Adding support for write() method to submit DSA work and changes to page fault handling

Will cherry-pick from 1.21.X to 1.22.X

### CODE REVIEW

File: prov/shm/src/smr_dsa.c [hunk 1]
```diff
@@ -184,6 +203,33 @@ static int dsa_open_wq(struct accfg_wq *wq)
...
+
+	// make sure writes (e.g., comp.status = 0) are ordered wrt to write()
+	{ asm volatile("sfence":::"memory"); }
+
+	ret = write(fd, &desc, sizeof(desc));
+
+	if (ret != sizeof(desc))
+		return false;
+
+	while (comp.status == 0 && max_comp_checks > 0) {
```
Comment (j-xiong) [inline prov/shm/src/smr_dsa.c:225]: Suggest use DSA_COMP_NONE here to avoid any confusion.  0 is often equivalent to SUCC in other definitions.

File: man/fi_shm.7.md [hunk 2]
```diff
@@ -163,12 +163,6 @@ The *shm* provider checks for the following environment variables:
 *FI_SHM_USE_DSA_SAR*
 : Enables memory copy offload to Intel DSA in SAR protocol. Default false
 
-*FI_SHM_ENABLE_DSA_PAGE_TOUCH*
-: Enables CPU touching of memory pages in a DSA command descriptor when the
-  page fault is reported, so that there is valid address translation for the
-  remaining addresses in the command. This minimizes DSA page faults. Default
-  false
-
 *FI_SHM_USE_XPMEM*
```
Comment (j-xiong) [inline man/fi_shm.7.md:166]: Please explain in the commit message why the changes are made.

**Comment by jcabr020** [comment]
@aingerson @vsanjeep

File: man/fi_shm.7.md (hunk 2 above)
Comment (jcabr020) [reply man/fi_shm.7.md:166]: Got it. Explained this, and also updated the code to use DSA_COMP_NONE. Pushed updates.

**Review by j-xiong — APPROVED** [review]

**Comment by j-xiong** [comment]
bot:aws:retest

**Comment by j-xiong** [comment]
bot:aws:retest

**Comment by wenduwan** [comment]
AWS CI failure on ARM
```
prov/shm/src/smr_dsa.c:38:10: fatal error: immintrin.h: No such file or directory
 #include <immintrin.h> // _mm_pause
          ^~~~~~~~~~~~~
compilation terminated.
make[1]: *** [prov/shm/src/src_libfabric_la-smr_dsa.lo] Error 1
make[1]: *** Waiting for unfinished jobs....
make: *** [all] Error 2
```

File: prov/shm/src/smr_dsa.c [hunk 3]
```diff
@@ -35,6 +35,7 @@
 #endif /* HAVE_CONFIG_H */
 
 #include "smr.h"
+#include <immintrin.h> // _mm_pause
```
Comment (j-xiong) [inline prov/shm/src/smr_dsa.c:38]: Please move this line to a place after `#if SHH_HAVE_DSA`, see the AWS CI failure when building on ARM.

**Review by j-xiong — CHANGES_REQUESTED** [review]

**Comment by j-xiong** [comment]
@wenduwan Thanks!

File: prov/shm/src/smr_dsa.c (hunk 3 above)
Comment (jcabr020) [reply prov/shm/src/smr_dsa.c]: Done. Pushed this update.

**Review by j-xiong — APPROVED** [review]
